src/index.js

```diff
@@ -1,42 +1,79 @@
-// ITERATION 1
-
 function updateSubtotal(product) {
   console.log('Calculating subtotal, yey!');
-
   //... your code goes here
+  let price = product.querySelector('.price span').innerHTML; //get price from direct child spans from class price.
+  //.innerHTML let's you access the content inside the span.
+
+  let quantity = product.querySelector('.quantity input').value; // get quantity from direct child input from class quantity.
+// You can access the attributes of the element with dot notation, or with the get Attributes
+  let subtotal = price * quantity; 
+
+  let subTotalPrice =  product.querySelector('.subtotal span') // Select span inside subtotal class.
+  subTotalPrice.innerHTML = subtotal; // use innerHTML to change the value of the selected content.
+  return subtotal;
 }
 
 function calculateAll() {
   // code in the following two lines is added just for testing purposes.
   // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
+  /* const singleProduct = document.querySelector('.product');
+  updateSubtotal(singleProduct); */
   // end of test
 
   // ITERATION 2
-  //... your code goes here
-
   // ITERATION 3
   //... your code goes here
+  let getProduct = document.getElementsByClassName('product');
+  let total = 0;
+/*   let getSubtotal = document.querySelector('.product .subtotal');
+  let getProductName = document.querySelector('.name span').innerHTML; */
+  
+  for (let i = 0; i < getProduct.length; i++) {
+    total += updateSubtotal(getProduct[i]);
+  }
+  document.querySelector('#total-value span').innerHTML = total;
+  return total;
 }
 
 // ITERATION 4
 
 function removeProduct(event) {
   const target = event.currentTarget;
-  console.log('The target in remove is:', target);
+  console.log('The target in remove is:', target.parentNode.parentNode);
+  target.parentNode.parentNode.remove();
   //... your code goes here
+/*   let getRemove = getElementsByClassName('btn-remove'); // Get an array-like object with all remove btns.
+
+  for (let i = 0; i < getRemove.length; i++) { // loop through all the remove btns
+    let addEvent = getRemove[i].addEventListener('click', removeProduct); // Add a click event to each remove btns
+    //getRemove[i].onclick = removeProduct(getRemove[i]);
+    let removeChild = target.getRemove.removeChild(getRemove[i]); // Remove the child
+    return removeChild;
+  } */
 }
 
 // ITERATION 5
 
 function createProduct() {
   //... your code goes here
```

Comment:
Here you're targeting the inputs and storing their values, targeting the table, creating a new <tr> element and trying to pass the values from the inputs to the new created element.
The logic behind is more or less what you've done, but you're missing some steps:

```
function createProduct() {
  // Targeting all the create product inputs that the user will fill
  let newProductInputs = document.querySelectorAll(".create-product input");

  // Save the value of the inputs in variables
  let newProductName = newProductInputs[0].value;
  let newProductPrice = newProductInputs[1].value;

  // Saving the newProduct html code in a variable
  let newProduct = `
  <tr class="product">
    <td class="name">
      <span>${newProductName}</span>
    </td>
    <td class="price">$<span>${newProductPrice}</span></td>
    <td class="quantity">
      <input type="number" value="0" min="0" placeholder="Quantity" />
    </td>
    <td class="subtotal">$<span>0</span></td>
    <td class="action">
      <button class="btn btn-remove">Remove</button>
    </td>
  </tr>`;

  // Targeting the table
  const tableBody = document.querySelector("tbody");

  // Adding the newProduct to the end of the table
  tableBody.insertAdjacentHTML("beforeend", newProduct);

  // Adding the removeProduct function to the remove button
  let removeBtn = document.querySelectorAll(".btn-remove");
  removeBtn.forEach((btn) => btn.addEventListener("click", removeProduct));
}
```